devdox/app/services/repository_service.py

```
@@ -99,3 +100,66 @@ async def get_all_provider_repos(
+                default_branch=transformed_data.default_branch,
+                forks_count=transformed_data.forks_count,
+                stargazers_count=transformed_data.stargazers_count,
+                is_private=transformed_data.private,
+                visibility=transformed_data.visibility,
+                size=transformed_data.size,
+                repo_created_at=transformed_data.repo_created_at,
+                language=languages,
+            )
+
+        except IntegrityError:
+            raise BadRequest(reason="Repository already exists")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)